githttp.go

```diff
@@ -231,7 +233,19 @@ func (g *GitHttp) getGitDir(file_path string) (string, error) {
 
 	f := path.Join(root, file_path)
 	if _, err := os.Stat(f); os.IsNotExist(err) {
-		return "", err
+		// If AutoCreate is false, just bail
+		if !g.AutoCreate {
+			return "", err
+		}
+		// If AutoCreate is true, attempt to create and initialise the directory
+		err = os.MkdirAll(f, os.ModePerm)
```

Comment:
It would be more readable and clear to write `0777` here instead of `os.ModePerm`.

Reply:
Apologies, this wasn't part of what I actually added but from another pull request (https://github.com/AaronO/go-git-http/pull/28).

If this (https://github.com/AaronO/go-git-http/pull/29/commits/35ca3c58d199869a91af62189eb3d8305671d913 is what I wrote) is actually going to be merged, I can fix this but it seems like there are lots of open pull requests not being merged.